### ISSUE TITLE
bad substitution not detected on broken bracket that follows a variable 

### ISSUE THREAD

**Issue by radkujawa** [koalaman/shellcheck#2142]
#### For bugs
- My shellcheck version: 0.7.1
- My bash version: 5.0.17

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

a="abc"
echo "${a[}"
```

#### Here's what shellcheck currently says:

nothing

#### Here's what I wanted or expected to see:

bad substitution: no closing `}' in "${a[}"
